Tow-Boot: update to -rc4 (rockchip + pinephonePro)

samueldr opened the pull request:
I'm misusing this to also handle https://github.com/Tow-Boot/U-Boot/pull/3

Fixes #297

* * *


Apparently I left a bunch of the work incomplete for the Pinephone Pro, as I hadn't verified that stuff like the buttons, LED and vibrator had their bindings.

With this change set, only those critical parts have been ported forward. We'll try not to stray too far from the upstream DT, and with the inevitable sync with mainline, we'll try to make it work even better.

Also adds back a change that removes an `iodomain` change, which breaks SPI Flash in e.g. the installer.

Finally, update a comment so I know why those config changes are required.

 - https://github.com/samueldr/u-boot/compare/tb-2023.07-007-rc3..wip/tb2307/ppp-dt

### What was done

 - Tested on Pinephone Pro, buttons, vibration and LED pattern started working again.

There is no difference for any other platform.

* * *

samueldr committed:
pine64-pinephonePro: Fixup workaround comment with symptoms

I removed it, thinking surely it was not needed anymore...
... it was.

L-as commented:
I tested this and it works fine, but I couldn't seem to boot from the SD card, albeit UMS works.

samueldr committed:
tow-boot: Mark 2023.07-007-rc4

https://github.com/Tow-Boot/U-Boot/compare/tb-2023.07-007-rc3...tb-2023.07-007-rc4

 - Adds missing Pinephone Pro changes
 - Increase reserved space for the kernel
